polyfill/lib/date.mjs

```diff
@@ -10,126 +10,126 @@ import {
   MILLISECOND,
   MICROSECOND,
   NANOSECOND,
+  CALENDAR,
   CreateSlots,
   GetSlot,
   SetSlot
 } from './slots.mjs';
 
+const ObjectAssign = Object.assign;
+
 export class Date {
-  constructor(isoYear, isoMonth, isoDay) {
+  constructor(isoYear, isoMonth, isoDay, calendar = undefined) {
     isoYear = ES.ToInteger(isoYear);
     isoMonth = ES.ToInteger(isoMonth);
     isoDay = ES.ToInteger(isoDay);
+    if (calendar === undefined) calendar = ES.GetDefaultCalendar();
     ES.RejectDate(isoYear, isoMonth, isoDay);
+    ES.RejectDateRange(isoYear, isoMonth, isoDay);
+    if (!calendar || typeof calendar !== 'object') throw new RangeError('invalid calendar');
```

Comment:
I know this isn't a new question, but, I guess `calendar` is an "exotic internal slot".  Should we check that the object is frozen?  What happens if someone mutates the calendar, affecting what is an otherwise immutable type?  Would it be safer to store a string ID instead of an object, and always do a lookup in the global registry?